R/http.R

```diff
@@ -352,11 +352,31 @@ httpRCurl <- function(protocol,
     error = function(e, ...) {
       # ignore errors resulting from timeout or user abort
       if (identical(e$message, "Callback aborted") ||
-          identical(e$message, "transfer closed with outstanding read data remaining"))
-        return
+          identical(e$message, "transfer closed with outstanding read data remaining")) {
+        return()
+      }
+
+      # detect errors due to overly-large bundles
+      if (grepl("long vectors not supported yet", e$message, fixed = TRUE)) {
+
+        msg <- "Deployment failed due to overly-large bundle."
+
+        prescription <- if (!identical(getOption("rsconnect.http"), "curl"))
+          "Try setting 'options(rsconnect.http = \"curl\")' and re-deploying your application."
```

Comment:
Good point; I've modified the prescription to only report `curl` as a candidate if `curl` is available on the PATH.